src/pkg/cluster/cluster.go

```diff
@@ -134,3 +154,204 @@ func WatcherForConfig(cfg *rest.Config) (watcher.StatusWatcher, error) {
 	sw := watcher.NewDefaultStatusWatcher(dynamicClient, restMapper)
 	return sw, nil
 }
+
+// InitOptions tracks the user-defined options during cluster initialization.
+type InitOptions struct {
+	// Indicates if Zarf was initialized while deploying its own k8s cluster
+	ApplianceMode bool
+	// Information about the repository Zarf is going to be using
+	GitServer types.GitServerInfo
+	// Information about the container registry Zarf is going to be using
+	RegistryInfo types.RegistryInfo
+	// Information about the artifact registry Zarf is going to be using
+	ArtifactServer types.ArtifactServerInfo
+	// StorageClass of the k8s cluster Zarf is initializing
+	StorageClass string
+}
+
+// Init takes initOptions and hydrates a cluster's state from InitOptions.
+func (c *Cluster) Init(ctx context.Context, opts InitOptions) error {
```

Comment:
IMO `Init`, `Load`, and `Save` should change to `InitState`, `LoadState`, and `SaveState`. Given that the cluster package deals with several different data types, it won't be super intuitive for users to know that these functions are dealing with Zarf state. 

For example, a user might be confused if the `cluster.New()` and `cluster.Save()` deal with separate data types.

Reply:
Yeah that is fair, and I agree that re-adding `State` into the cluster methods that return a `State` is a better API than what exists in the PR now. In future iterations I wonder whether we can't simplify `cluster` even more - `New`, `InitState`, and `LoadState` all feel like similar operations... not sure yet, much to consider.